examples/mass_delete_old_medias.py

```diff
@@ -0,0 +1,460 @@
+                    )
+                    to_fallback += medias_per_speaker[recipient]
+                else:
+                    logger.debug(
+                        f'Sent "{recipient}" an email about {len(medias_per_speaker[recipient])} medias.'
+                    )
+                    sent_count += 1
+            if to_fallback:
+                fallback_message = _prepare_mail(
+                    msc,
+                    sender_address,
+                    fallback_email,
+                    to_fallback,
+                    delete_date,
+                    skip_category,
+                    html_template,
+                    plain_template,
+                    email_subject_template,
+                )
+                smtp.sendmail(sender_address, fallback_email, fallback_message)
+                sent_count += 1
+        logger.info(f'Sent {sent_count} emails.')
+    else:
+        for speaker_email, speaker_medias in medias_per_speaker.items():
+            logger.debug(
+                f'[Dry run] Would have sent "{speaker_email}" an email about {len(speaker_medias)} medias.'
+            )
+        sent_count = len(to_send) + (1 if to_fallback else 0)
+        logger.info(
+            f'[Dry run] {sent_count} emails would have been sent.'
+        )
+
+
+def _delete_medias(msc: MediaServerClient, medias: list[dict], apply: bool = False):
+    ms_url = msc.conf['SERVER_URL'] + '/permalink/'
+    medias = {media['oid']: media for media in medias}
+    deleted_count = 0
+    deleted_size = 0
+    if apply:
+        response = msc.api(
+            'catalog/bulk_delete/',
+            method='post',
+            data=dict(oids=list(medias.keys()))
+        )
+        for oid, result in response['statuses'].items():
+            if result['status'] == 200:
+                logger.debug(f'Media {ms_url}{oid} has been deleted.')
+                deleted_count += 1
+                deleted_size += medias[oid]['storage_used']
+            else:
+                err = result['message']
+                logger.error(
+                    f'An error occurred while attempting to delete media {ms_url}{oid}. '
+                    f'The media has not been deleted: {err}.'
+                )
+        logger.info(
+            f'{deleted_count} medias ({pp_size(deleted_size)}) have been successfully deleted.'
+        )
+    else:
+        for oid, media in medias.items():
+            logger.debug(f'[Dry run] Media {ms_url}{oid} would have been deleted.')
+            deleted_count += 1
+            deleted_size += media['storage_used']
+        logger.info(
+            f'[Dry run] {deleted_count} medias ({pp_size(deleted_size)}) would have been have been deleted.'
+        )
+
+
+class ArgumentParser(argparse.ArgumentParser):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.add_argument(
+            '--conf',
```

Comment:
Why do we need to subclass ArgumentParser ?

Reply:
Because I prefer to print the help on parsing error: https://github.com/UbiCastTeam/mediaserver-client/pull/26/files#diff-0d45181989f23fd6dbd15ca1c675057596a69d01a27339fedbcda6ec8e5d33a2R377